### PR TITLE
Split out ast-builders into multiple utils

### PR DESCRIPTION
**Pull request by MonicaOlejniczak**
## Changes
* Add `getJSXAttribute` util
* Create `Variable` type
* Extract `buildCompiledComponent`, `buildDisplayName`, `compiledTemplate`, and `hoistSheet` to separate utils
* Move `importSpecifier` to `append-runtime-imports.ts`
* Rename `buildCssVariablesProp` to `buildCssVariables` and extract it to a separate util
* Rename `transformItemCss` to `transformCssItems` and extract it to a separate util

### CODE REVIEW

File: packages/babel-plugin/src/styled/index.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import * as t from '@babel/types';
 import type { NodePath } from '@babel/core';
+
+import type { Metadata, Tag } from '../types';
```
Comment (madou) [inline packages/babel-plugin/src/styled/index.tsx:4]: Can we turn a lint rule on for imports?

Comment (MonicaOlejniczak) [reply packages/babel-plugin/src/styled/index.tsx:4]: For sorting and grouping? I was planning to, otherwise not sure what you're referring to

Comment (madou) [reply packages/babel-plugin/src/styled/index.tsx:4]: Yeah exactly

**Review by madou — DISMISSED** [review]
LGTM

**Pull request merged**